### ISSUE TITLE
val acc is higher than train acc

### ISSUE THREAD

**Issue by qiangwang57** [yaysummeriscoming/DALI_pytorch_demo#3]
Hi  @yaysummeriscoming,

Thanks for the amazing work, which is really helpful! I used your code to train on imagenet subset with 120 classes, but I found that val acc is always higher than train acc, no matter the subset the original one. Below is the output on the subset for 5 epochs:

```
defaults for this optimization level are:
enabled                : True
opt_level              : O1
cast_model_type        : None
patch_torch_functions  : True
keep_batchnorm_fp32    : None
master_weights         : None
loss_scale             : dynamic
Processing user overrides (additional kwargs that are not None)...
After processing overrides, optimization options are:
enabled                : True
opt_level              : O1
cast_model_type        : None
patch_torch_functions  : True
keep_batchnorm_fp32    : None
master_weights         : None
loss_scale             : dynamic
Epoch=1/5, lr: 0.02000, train=[top1: 3.2319, top5: 12.1266, loss: 4.4690, time: 194.651, speed: 595.4], val=[top1: 8.0400, top5: 24.3900, loss: 4.1161, time: 11.691, speed: 829.7]
Epoch=2/5, lr: 0.04000, train=[top1: 10.2839, top5: 29.5866, loss: 3.9342, time: 194.551, speed: 595.7], val=[top1: 16.2100, top5: 40.1800, loss: 3.5553, time: 10.180, speed: 952.8]
Epoch=3/5, lr: 0.06000, train=[top1: 17.6307, top5: 42.9325, loss: 3.4815, time: 194.996, speed: 594.3], val=[top1: 26.8300, top5: 55.3400, loss: 2.9659, time: 10.182, speed: 952.6]
Epoch=4/5, lr: 0.08000, train=[top1: 24.2679, top5: 52.3032, loss: 3.1226, time: 194.863, speed: 594.7], val=[top1: 31.1200, top5: 60.2300, loss: 2.7813, time: 10.250, speed: 946.4]
Epoch=5/5, lr: 0.10000, train=[top1: 30.4317, top5: 59.8794, loss: 2.8106, time: 195.180, speed: 593.8], val=[top1: 36.2000, top5: 65.5700, loss: 2.5313, time: 10.270, speed: 944.5]
```

Any comments?

Best wishes, Chiang

**Comment by yaysummeriscoming** [comment]
Great to hear.  Are you training with the small shufflenet V2? In which case, I would expect this behaviour.  This is a very small model and models using separable convolutions have far less parameters, meaning they tend to overfit less.  The preprocessing augmentations (primarily the random resized crop) can therefore reduce train accuracy below validation accuracy.

**Comment by qiangwang57** [comment]
Nope. I tried resnet18, 50, and 101, all present similar results. For data augmentation, I added color jitter upon the default ones. The training and val loss are reasonable though, i.e., val loss is less than training loss.

cheers, Chiang

**Comment by yaysummeriscoming** [comment]
Ok, have you tried training on the complete dataset?

**Comment by qiangwang57** [comment]
Yup, on both full and sub imagenet, but similar.

**Comment by yaysummeriscoming** [comment]
Hmm I had a look back at a reference run I did of resnet18 on the entire ImageNet dataset:
![Model Accuracy](https://user-images.githubusercontent.com/11413145/80183936-4f9bed00-863c-11ea-84f0-4a5637d8b4ba.png)

First epochs:
<img width="601" alt="Screen Shot 2020-04-24 at 9 01 04 am" src="https://user-images.githubusercontent.com/11413145/80184014-70644280-863c-11ea-90c4-4b87117f401f.png">

Due to the small minimum crop size (8%), the train accuracy can be lower than the validation accuracy - but this is normal.  I'd suggest to do a run using the DALI reference example [here](https://github.com/NVIDIA/DALI/blob/master/docs/examples/use_cases/pytorch/resnet50/main.py) and confirming you meet the expected accuracy.
